tools/monorepo-utils/src/code-freeze/commands/changelog/lib/index.ts

```diff
@@ -242,21 +243,157 @@ export const updateTrunkChangelog = async (
+			Logger.notice(
+				`No commits between ${ releaseBranch } and the branch, skipping the PR.`
+			);
+		} else if (
+			e.message.includes( 'did not match any file(s) known to git' )
+		) {
 			Logger.notice(
-				'No commits between trunk and the branch, skipping the PR.'
+				`Branch ${ releaseBranch } does not exist, skipping the PR.`
 			);
 		} else {
 			Logger.error( e );
 		}
 	}
 };
+
+/**
+ * Perform changelog operations on trunk by submitting a pull request.
+ *
+ * @param {Object} options                                 CLI options
+ * @param {string} tmpRepoPath                             temp repo path
+ * @param {Object} releaseBranchChanges                    update data from updateReleaseBranchChangelogs
+ * @param {Object} releaseBranchChanges.deletionCommitHash commit from the changelog deletions in updateReleaseBranchChangelogs
+ * @param {Object} releaseBranchChanges.prNumber           pr number created in updateReleaseBranchChangelogs
+ */
+export const updateTrunkChangelog = async (
+	options: Options,
+	tmpRepoPath: string,
+	releaseBranchChanges: { deletionCommitHash: string; prNumber: number }
+): Promise< void > => {
+	return await updateBranchChangelog(
+		options,
+		tmpRepoPath,
+		'trunk',
+		releaseBranchChanges
+	);
+};
+
+/**
+ * Retrieves the WooCommerce version from the trunk branch
+ *
+ * @param tmpRepoPath cloned repo path
+ * @return the WooCommerce version string if found, or `null` if not found.
+ */
+async function getTrunkWooCommerceVersion(
+	tmpRepoPath: string
+): Promise< string | null > {
+	const git = simpleGit( {
+		baseDir: tmpRepoPath,
+		config: [ 'core.hooksPath=/dev/null' ],
+	} );
+
+	await git.checkout( 'trunk' );
+
+	const wooCommercePhpPath = path.join(
+		tmpRepoPath,
+		'plugins/woocommerce/woocommerce.php'
+	);
+	const fileContent = readFileSync( wooCommercePhpPath, 'utf8' );
+
+	const versionMatch = fileContent.match( /\*\s+Version:\s+(\d+\.\d+)/ );
+	const version = versionMatch ? versionMatch[ 1 ] : null;
+
+	Logger.notice( `WooCommerce trunk version is ${ version }` );
+
+	return version;
+}
+
+/**
+ * Generates a list of release branch names between the target version and the trunk version.
+ * Each branch name follows the format `release/{major}.{minor}`.
+ *
+ * @param trunkVersion  the current trunk version in the "major.minor" format (e.g., "8.7").
+ * @param targetVersion the target version in the "major.minor" format (e.g., "9.5").
+ * @return An array of branch names representing all release branches between the target and trunk versions.
+ */
+function getTargetBranches(
+	trunkVersion: string,
+	targetVersion: string
+): string[] {
+	const [ targetMajor, targetMinor ] = targetVersion
+		.split( '.' )
+		.map( Number );
+	const [ trunkMajor, trunkMinor ] = trunkVersion.split( '.' ).map( Number );
+
+	const branches: string[] = [];
+
+	for (
+		let v = Number( `${ targetMajor }.${ targetMinor }` ) + 0.1;
+		v < Number( `${ trunkMajor }.${ trunkMinor }` );
```

Comment:
The trunk version is not going to change, so we can also move it into a local variable.
Like `for (let current = ..., limit = ...; current < limit; current += ...) {`

Reply:
I agree with the bot that using floats in this way is risky. It's smart, though, so let's keep it but add some safety measures here:
- tests for this function
- code hardening as (pseudo-code): `v.toFixed( 1 ) < Number( `${ trunkMajor }.${ trunkMinor }` ).toFixed( 1 )`

Reply:
Thanks for the review! I think you are right, I've been doing some more testing and I see some rounding errors, it seems brittle indeed. I've pushed a different solution in https://github.com/woocommerce/woocommerce/pull/59562/commits/b00d5eb5b2d7464111eec57895fc54e3c85a25c2, let me know what you think 🙏